omniduct/filesystems/s3.py

```diff
@@ -189,11 +191,73 @@ def _dir(self, path):
     #         if pattern is None or pattern.match(k.key[len(path_prefix):]):
     #             yield k.key
 
+    def _format_s3_path(self, path):
+        return 's3:' + path[4:] if path.startswith('s3n:') or path.startswith('s3a:') else path
+
+    def _is_s3(self, path):
+        """This function gives True if the given path is valid s3 path. This is used for the purpose of
+        validation."""
+        return 's3' in self._format_s3_path(path)
+
+    def _bucket_name(self, path):
+        """This function gives the bucket name.
+        Example: If the path s3://bucket_name/folder1/folder2/ is given as a parameter to this function
+        it returns bucket_name"""
+        if self._is_s3(path):
+            path = urlparse(path)
+            return path.netloc
+        else:
+            raise FileNotFoundError("path `{}` does not exist.".format(path))
+
+    def _file_or_dirpath(self, path):
```

Comment:
Paths are already expected to be of the form output by this function, as path sanitisation is handled by the parent `FileSystemClient` class.

Reply:
didn't get this comment. Can you please elaborate?

Reply:
The `path` variable passed to most methods starting with an _ (including _mkdir and _remove) has already been parsed and checked, and hence do not need to be reparsed. They are already of the form output by your new method.